lib/index.js

```diff
@@ -418,6 +420,16 @@ Segment.prototype.retrieveCrossDomainId = function(callback) {
   }
 };
 
+function mergeBlockedTraits(msg) {
+  var options = msg.options() || {};
+  var json = msg.json() || {};
+  if (options.blockedTraits && json.traits) {
+    extend(json.traits, options.blockedTraits);
```

Comment:
Can there be blocked traits but `json.traits` is null?

Reply:
I don't see how that would be possible since the `blockedTraits` come from the `json.traits` object?

Reply:
It may be depending on how we remove the blocked traits upstream (i.e. whether we set empty or null when all the traits get blocked).

Reply:
I'm just deleting properties in `analytics.js-core`, so when all traits get blocked `traits` will just be an empty object.